Sources/ConnectionPoolModule/PoolStateMachine+ConnectionGroup.swift

```diff
@@ -547,8 +562,9 @@ extension PoolStateMachine {
 
             if closedAction.wasRunningKeepAlive {
                 self.stats.runningKeepAlive -= 1
+            } else {
```

Comment:
this change looks wrong, do you mind explaining, why you changed this one?

Reply:
Yeah, that's wrong. I did this initially because a test I've added force closed the connection without going the proper route. This caused a arythmic overflow due to leasedStreams being 0 at that point in time. I've fixed the test and reverted the change.